lib/proxymode.pm

```diff
@@ -0,0 +1,142 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+package proxymode;
+# Summary: proxymode: The basic lib for using proxy mode to connect or operation with physical machine
+# Maintainer: John <xgwang@suse.com>
+
+use base 'basetest';
+use testapi;
+use strict;
+
+our $SLAVE_SERIALDEV = 'proxyserial';
+
+sub switch_power {
+    my ($ipmi_machine, $ipmi_user, $ipmi_pass, $ipmi_status) = @_;
+    $ipmi_pass   //= 'ADMIN';
+    $ipmi_user   //= 'ADMIN';
+    $ipmi_status //= 'off';
+    die "Variable ipmi_machine is invalid in function restart_host!" unless $ipmi_machine;
+    my $ipmitool = "ipmitool -H " . $ipmi_machine . " -U " . $ipmi_user . " -P " . $ipmi_pass . " -I lanplus ";
+    select_console 'log-console';
```

Comment:
Switching console before restarting machine is only for fixing issue about old machine with ipmi version 1.7, On ipmi version 1.7, i need to connect machine at first , then restart machine (otherwise the connection of ipmi usually happens error, i do not know the reason yet), and the connection of ipmi is interactive mode, So i can only switch another console for that.